test/reline/test_reline.rb

```diff
@@ -331,6 +334,7 @@ def test_add_dialog_proc
     assert_raise(ArgumentError) { Reline.add_dialog_proc(:error, 42) }
     assert_raise(ArgumentError) { Reline.add_dialog_proc(:error, 'hoge') }
     assert_raise(ArgumentError) { Reline.add_dialog_proc('error', proc {} ) }
+    assert_raise(ArgumentError) { Reline.add_dialog_proc('error', nil) }
```

Comment:
I think either is fine as a specification.

I found an example `ENV[:a] = 1` raise error and `ENV.delete(:a)` raise error.
But there is another example `array[:a] = 1` raise error but `array.delete(:a)` does not raise error.

One good thing is that it might be easier to debug if removing non-string name raise error for the code below
```ruby
NAME = :invalid_symbol_name
def disable_dialog() = Reline.add_dialog_proc(NAME, nil)
def enable_dialog() = Reline.add_dialog_proc(NAME, PROC)
disable_dialog # Argument Error
```

Reply:
I'm asking simply because the previous case (`Reline.add_dialog_proc('error', proc {} )`) probably already covered the same code path?

Reply:
I see, It's covered. I removed it 👍